test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -1,7 +1,261 @@
+	ensureClusterInGoodState(t, cs, 10*time.Second, 6*time.Minute, 1*time.Minute)
+
+	// cause a disruption
+	data := getDefaultUnsupportedConfig(t, cs)
+	//data["apiServerArguments"] = map[string]string{"non-existing-flag": "true"}
+	setUnsupportedConfig(t, cs, data)
+
+	// validate if the fallback condition is reported and the cluster is stable
+	waitForFallbackDegradedCondition(t, cs)
+	nodeName, failedRevision := assertNodeStatus(t, cs)
+	assertKasPodAnnotatedOnNode(t, cs, failedRevision, "*", "*", nodeName)
+	// TODO: try kas - get pod/ns ?
+	// TODO: check if pod is ready ?
+
+	// clean up
+	setUnsupportedConfig(t, cs, getDefaultUnsupportedConfig(t, cs))
+	if err := clusterInGoodState(t, cs, 10*time.Second, 6*time.Minute, 1*time.Minute); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func ensureClusterInGoodState(t testing.TB, cs clientSet, waitPollInterval, waitPollTimeout, mustBeReadyFor time.Duration) {
+	if err := clusterInGoodState(t, cs, waitPollInterval, waitPollTimeout, mustBeReadyFor); err != nil {
+		t.Logf("Trying to bring the cluster back to normal state by removing all unsupportedConfigOverrides entries, err: %v", err)
+		setUnsupportedConfig(t, cs, getDefaultUnsupportedConfig(t, cs))
+		err := clusterInGoodState(t, cs, waitPollInterval, waitPollTimeout, mustBeReadyFor)
+		require.NoError(t, err)
+	}
+}
+
+func clusterInGoodState(t testing.TB, cs clientSet, waitPollInterval, waitPollTimeout, mustBeReadyFor time.Duration) error {
+	t.Helper()
+
+	startTs := time.Now()
+	t.Logf("Waiting %s for the cluster to be in a good condition, interval = %v, timeout %v", mustBeReadyFor.String(), waitPollInterval, waitPollTimeout)
+
+	return wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		monitorPod, err := cs.Kube.CoreV1().Pods("openshift-kube-apiserver").Get(context.TODO(), "kube-apiserver-startup-monitor", metav1.GetOptions{})
+		if err == nil {
+			t.Logf("the static pod monitor: %v is running since: %v", monitorPod.Name, monitorPod.CreationTimestamp)
+			return false, nil /*retry*/
+		} else if !errors.IsNotFound(err) {
+			t.Logf("error while getting kube-apiserver-startup-monitor: %v", err)
+			return false, nil /*retry*/
+		}
+
+		kasPods, err := cs.Kube.CoreV1().Pods("openshift-kube-apiserver").List(context.TODO(), metav1.ListOptions{LabelSelector: "apiserver=true"})
+		if err != nil {
+			t.Logf("error while getting kube-apiserver pods: %v", err)
+			return false, nil /*retry*/
+		}
+		if len(kasPods.Items) == 0 {
+			t.Log("expected to find the kube-apiserver static pod but haven't found any")
+			return false, nil /*retry*/
+		}
+		for _, kasPod := range kasPods.Items {
+			if fallbackFor, ok := kasPod.Annotations[annotations.FallbackForRevision]; ok {
+				t.Logf("%v pod has %v annotation set to: %v", kasPod.Name, annotations.FallbackForRevision, fallbackFor)
+				return false, nil /*retry*/
+			}
+		}
+		// TODO: on an HA cluster we could also check if pods are on the same revision
```

Comment:
I would consider a cluster no in-progress as good (i.e. nodeStatus has no transitioning pod). Why do we need all of this?

Reply:
just playing safe, checking if `nodeStatus` contain only `currentRevision` and `nodeName` would be enough?

Reply:
I think it is enough to start with. We also don't want to start from some "dirty" state.